docs/CODING_STYLE.md

```diff
@@ -152,25 +152,35 @@ layout: default
 ## Using C Constructs
 
 - Allocate local variables where it makes sense: at the top of the block, or at
-  the point where they can be initialized. `r` is typically used for a local
-  state variable, but should almost always be declared at the top of the
+  the point where they can be initialized. Avoid huge variable declaration
+  lists at the top of the function. In long functions that require many
+  variables, consider moving some of the declarations close to the point of
+  first use. As an exception, `r` is typically used for a local state variable,
+  but should almost always be declared as the last variable at the top of the
   function.
```

Comment:
hmm, my suggestion would be to split these function up, so that the variable declarations and the uses are not that far apart from each other.

I am not convinced going from one extreme (everything on top) to the other (everything where used) is really the best choice, i'd just allow both and be done with it for now. And smaller functions are definitely a good idea in the majority of cases I think, a much better one than placing variable definitions in the middle of functions I think.